modules/packages/arch/metadata.go

```diff
@@ -43,8 +43,9 @@ var (
 	ErrInvalidArchitecture = util.NewInvalidArgumentErrorf("package architecture is invalid")
 
 	// https://man.archlinux.org/man/PKGBUILD.5
-	namePattern    = regexp.MustCompile(`\A[a-zA-Z0-9@._+-]+\z`)
-	versionPattern = regexp.MustCompile(`\A(?:[0-9]:)?[a-zA-Z0-9.+~]+(?:-[a-zA-Z0-9.+-~]+)?\z`)
+	namePattern = regexp.MustCompile(`\A[a-zA-Z0-9@._+-]+\z`)
+	// (epoch:pkgver-pkgrel)
+	versionPattern = regexp.MustCompile(`\A(?:[0-9]:)?[a-zA-Z0-9._+~]+(?:-[a-zA-Z0-9.+-_~]+)?\z`)
```

Comment:
```suggestion
	versionPattern = regexp.MustCompile(`\A(?:[\d]:)?[\w.+~]+(?:-[\w.+~-]+)?\z`)
```

Maybe it could have some tests